docs/infrastructure/logging.md

```diff
@@ -387,3 +398,62 @@ For example, in the above example `project-manager` will log events of up-to
 
 **Note** Logging to a file requires presence of the `file`
 [appender](#file-appender) in the `logging-service.appenders` section.
+
+# How to use logging
+
+Logging infrastructure uses a popular SLF4J interface which most of developers
+should be familiar with. In this section we include a only small number of
+examples, full user manual is available at SLF4J's
+[website](https://www.slf4j.org/manual.html).
+
+## Log a simple INFO message
+
+```
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HelloWorld {
+
+  public static void main(String[] args) {
+    Logger logger = LoggerFactory.getLogger(HelloWorld.class);
+    logger.info("Hello World");
+  }
+}
+```
+
+## Log a simple INFO message only if TRACE is enabled
+
+```
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HelloWorld {
+
+  public static void main(String[] args) {
+    Logger logger = LoggerFactory.getLogger(HelloWorld.class);
+    if (logger.isTraceEnabled()) {
+      logger.info("Hello World");
+    }
```

Comment:
Is this the right pattern? I would expect simple `logger.trace("Hello World")`. How do I need to launch the CLI for such a message to appear in the console?

Reply:
You asked for `log a message into console only if verbosity is increased`. Of course one can use `.trace` but that's different from your request.

Reply:
> You asked for `log a message into console only if verbosity is increased`. Of course one can use `.trace` but that's different from your request.

I have asked for a way to _"log a message to a console if a verbosity is increased"_. Does:
```java
logger.trace("Hello World")
```
appear on the console when `enso` is invoked (and the `logger` code is performed)? Does the `Hello World` message appear on the console when `enso` is invoked with some logging parameters? If so, then `logger.trace` + description of the CLI arguments to enable seeing the message on console is the solution.

PS: Using `isTraceEnabled()` is probably only good for performance reasons, isn't it? If so, then your example should be:
```java
    Logger logger = LoggerFactory.getLogger(HelloWorld.class);
    if (logger.isTraceEnabled()) {
      logger.trace("Hello World");
    }
```
e.g. using `isTraceEnabled` and `.trace`. Combining it with `.info` is pretty weird from my perspective.